src/components/ActionSheetItem/ActionSheetItem.tsx

```diff
@@ -40,6 +40,10 @@ export interface ActionSheetItemProps
    */
   onClick?: React.MouseEventHandler<HTMLElement>;
   onImmediateClick?: React.MouseEventHandler<HTMLElement>;
+  /**
+   * Иконка для `checked` режима.
+   */
+  iconChecked?: React.ReactNode;
```

Comment:
К этой теме #3411

Мб вот так

```suggestion
  iconChecked?: React.ReactNode | {
    compact: React.ReactNode;
    regular: React.ReactNode;
  };
```

в v5 внутри компонента так и так надо будет делать так:

```jsx
<SizeXConditionalRender
  compact={<Icon20CheckCircleOn aria-hidden />}
  regular={<Icon24CheckCircleOn aria-hidden />}
/>
```

Если передали без `compact` и `regular`, то не используем `SizeXConditionalRender`

---

PS1: кстати, забавно, что всё ещё непонятно `SizeXConditionalRender` использовать или `SizeYConditionalRender`
PS2: и мне, кажется, `ComponentType` всё же будет красивей в этом кейсе, т.к. мы переопределяем базовую иконку

Reply:
Думал сделать так

```
iconChecked?: React.ReactNode = <SizeYConditionalRender
  compact={<Icon20CheckCircleOn aria-hidden />}
  regular={<Icon24CheckCircleOn aria-hidden />}
/>
```

Но это все в рамках https://github.com/VKCOM/VKUI/issues/3411

> PS2: и мне, кажется, ComponentType всё же будет красивей в этом кейсе, т.к. мы переопределяем базовую иконку

А если мы используем иконку не из `@vkontakte/icons`? Есть реальные кейсы

Reply:
> Почему нет

В иконке могут отсутствовать свойства, которые прописаны в типе

Или тот же вариант, когда хочешь запихнуть туда SizeYConditionalRender

Reply:
> В иконке могут отсутствовать свойства, которые прописаны в типе

ну вот плохо если будут отсутствовать

пользователь закидывает кастомную иконку и должен обеспечить тот же интерфейс, который требуется

Reply:
Но по сути нам и не нужен тут интерфейс от иконки